fix(npm/pnpm): revert "fix(npm/pnpm): redirect cache and store dir" to attempt to fix `pnpm dedupe` hanging

karlhorky opened the pull request:
Reverts renovatebot/renovate#35807

As per the discussion in https://github.com/renovatebot/renovate/discussions/36336 and https://github.com/renovatebot/renovate/discussions/35840 and https://github.com/renovatebot/renovate/discussions/36006 and https://github.com/renovatebot/renovate/discussions/36143 , Renovate is currently broken for pnpm due to `pnpm dedupe` hanging on stdin due to a `store-dir` misalignment.

"Broken" in this case can mean either:

1. ["errors in PRs"](https://github.com/phi-ag/ifc-guid/pull/25) ([example 2](https://github.com/phi-ag/rvt-app/pull/428)) - with `temporary-error` explanation (from @peterhirn's report in   https://github.com/renovatebot/renovate/discussions/36006#discussioncomment-13389181)
2. ["silently failing to create new PRs"](https://github.com/upleveled/portfolio-cms/issues/22) ([example 2](https://github.com/upleveled/preflight/issues/93), [example 3](https://github.com/upleveled/next-starter-peacock/issues/34), [example 4](https://github.com/karlhorky/chatgpt-summary-browser-extension/issues/4)) (from my report in https://github.com/renovatebot/renovate/discussions/36336#discussioncomment-13487253) (see screenshot below, showing that I checked the Dependency Dashboard checkboxes to trigger an upgrade 15 hours ago, and they stay checked, with no visible errors for the user in GitHub)

![Screenshot 2025-06-17 at 09 17 56](https://github.com/user-attachments/assets/b899d4df-da3b-4952-a18b-91e251b02ae2)

Either way, it blocks usage of Renovate on pnpm repos - maybe specifically, pnpm repos which use the `packageManager` field in `package.json`.

What causes `pnpm dedupe` to be broken in this way? @dipunm's note from https://github.com/renovatebot/renovate/discussions/35840 seems helpful:

> Specifically, I found that after execing into the running docker image and cding to the repository path:
>
> - `pnpm store path` responded with `/home/renovate/.local/share/pnpm/store/v10`
> - `cat node_modules/.modules.yaml` shows: `storeDir: /tmp/renovate/cache/others/pnpm-store/v10`
>
> In the version before, I can see that:
> 
> - `pnpm store path` responded with `/home/renovate/.local/share/pnpm/store/v10`
> - `cat node_modules/.modules.yaml` shows: `storeDir: /home/renovate/.local/share/pnpm/store/v10`

How can we resolve it? From reading the discussions, it appears that @CHC383 has a good suggestion [here](https://github.com/renovatebot/renovate/discussions/36336#discussioncomment-13487753):

> we should either revert #35807 to stay on default, or also set the same path on pnpm (`pnpm config set store-dir`)

This unfortunately reverts renovatebot/renovate#35807, but it looks like that PR broke pnpm compatibility.

Happy to adjust to change this.

Looking forward to having a non-broken Renovate experience with pnpm!

### Reproduction

Minimal reproduction for `2. "silently failing to create new PRs"` here:

- https://github.com/karlhorky/repro-renovate-pnpm-dedupe-hanging

![Screenshot 2025-06-18 at 11 48 50](https://github.com/user-attachments/assets/8c2752b6-77a6-470d-a6ba-c3ec7eccb556)

![Screenshot 2025-06-18 at 11 52 58](https://github.com/user-attachments/assets/282a3a12-f8d7-47b7-90ea-8ba56ecb0f26)

cc @viceice @rarkins @peterhirn @dipunm @CHC383

karlhorky committed:
Revert "fix(npm/pnpm): redirect cache and store dir (#35807)"

This reverts commit 8fb6a9c89e7c293895c5a4ae861596c97601a189.

viceice reviewed (changes requested):
I need to investigate deeper. I'm also using pnpm with dedupe and don't see that issue 🤔

karlhorky commented:
It appears that @peterhirn is also configuring `postUpdateOptions: ['pnpmDedupe']`

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    ":semanticCommits",
    "config:best-practices",
    "group:monorepos",
    "group:recommended"
  ],
  "postUpdateOptions": ["pnpmDedupe"],
  "lockFileMaintenance": {
    "enabled": true,
    "automerge": true
  },
  "packageRules": [
    { "matchDepNames": ["node"], "ignoreUnstable": false },
    {
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "matchCurrentVersion": "!/^0/",
      "automerge": true
    }
  ]
}
```

Source: https://github.com/phi-ag/ifc-guid/blob/17d41ee699ea03bf2fe17fa7f215e430e0336d14/renovate.json#L9

viceice commented:
can you create a minimal reproduction please?

viceice commented:
can you create a minimal reproduction please?

karlhorky committed:
Merge branch 'main' into revert-35807-fix/pnpm/cache-dir

karlhorky commented:
@viceice I checked the Dependency Dashboard of the reproduction just now, and I see that it is still hanging (screenshots above - I toggled the checkbox in the Dependency Dashboard yesterday)

The main parts of the reproduction:

`renovate.json`

```json
{
  "extends": ["config:base"],
  "postUpdateOptions": ["pnpmDedupe"],
  "packageRules": [
    {
      "groupName": "dependency upgrades - non-major",
      "matchDepTypes": ["devDependencies", "packageManager"],
      "updateTypes": ["patch", "minor"]
    }
  ]
}
```

`package.json`

```json
{
  "devDependencies": {
    "stylelint": "16.19.1"
  },
  "packageManager": "pnpm@10.10.0+sha512.d615db246fe70f25dcfea6d8d73dee782ce23e2245e3c4f6f888249fb568149318637dca73c2c5c8ef2a4ca0d5657fb9567188bfab47f566d1ee6ce987815c39"
}
```

<details>

<summary>Show logs</summary>

```
DEBUG: Using RE2 regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /runner/renovate/job_config.json
DEBUG: Detected config in env RENOVATE_CONFIG
{
  "config": {
    "extends": [
      "mergeConfidence:all-badges"
    ],
    "prFooter": "This PR was generated by [Mend Renovate](https://mend.io/renovate/). View the [repository job log](https://developer.mend.io/{{platform}}/{{repository}}).",
    "redisUrl": "redis://developer-platform-bare-metal-prd.aqffol.ng.0001.use1.cache.amazonaws.com:6379",
    "binarySource": "install",
    "globalExtends": [
      "global:safeEnv"
    ],
    "logLevelRemap": [
      {
        "newLogLevel": "info",
        "matchMessage": "Cache fallback URLs"
      }
    ],
    "platformCommit": "enabled",
    "allowedCommands": [
      "^git add --all$",
      "^git reset$",
      "^pwd$"
    ],
    "repositoryCache": "enabled",
    "cacheTtlOverride": {
      "datasource-docker-hub-tags": 90,
      "datasource-maven:metadata-xml": 90
    },
    "encryptedWarning": "Encrypted secrets were detected in the repository config. This feature has been deprecated. Please migrate them to this portal. See the guide at http://docs.renovatebot.com/mend-hosted/migrating-secrets/",
    "onboardingConfig": {
      "$schema": "https://docs.renovatebot.com/renovate-schema.json",
      "extends": [
        "config:recommended"
      ]
    },
    "gitIgnoredAuthors": [
      "29139614+renovate[bot]@users.noreply.github.com"
    ],
    "customizeDashboard": {
      "repoProblemsHeader": "These problems occurred while renovating this repository. [View logs](https://developer.mend.io//{{platform}}/{{repository}})."
    },
    "repositoryCacheType": "local",
    "dependencyDashboardFooter": "\n- [ ] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n",
    "dependencyDashboardHeader": "This issue lists Renovate updates and detected dependencies. Read the [Dependency Dashboard](https://docs.renovatebot.com/key-concepts/dashboard/) docs to learn more.<br>[View this repository on the Mend.io Web Portal](https://developer.mend.io/{{platform}}/{{repository}})."
  }
}

DEBUG: File config
{
  "config": {
    "token": "***********",
    "hostRules": [
      {
        "hostType": "docker",
        "matchHost": "docker.io",
        "username": "mdpprod2renovate",
        "password": "***********"
      },
      {
        "hostType": "docker",
        "matchHost": "hub.docker.com",
        "username": "mdpprodrenovatedc",
        "password": "***********"
      },
      {
        "hostType": "github",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "github-tags",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "github-releases",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "pod",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "terraform-provider",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "rubygems",
        "matchHost": "github.com",
        "token": "***********"
      }
    ],
    "secrets": {}
  }
}

DEBUG: CLI config
{
  "config": {}
}

DEBUG: Env config
{
  "config": {
    "extends": [
      "mergeConfidence:all-badges"
    ],
    "prFooter": "This PR was generated by [Mend Renovate](https://mend.io/renovate/). View the [repository job log](https://developer.mend.io/{{platform}}/{{repository}}).",
    "redisUrl": "redis://developer-platform-bare-metal-prd.aqffol.ng.0001.use1.cache.amazonaws.com:6379",
    "binarySource": "install",
    "globalExtends": [
      "global:safeEnv"
    ],
    "logLevelRemap": [
      {
        "newLogLevel": "info",
        "matchMessage": "Cache fallback URLs"
      }
    ],
    "platformCommit": "enabled",
    "allowedCommands": [
      "^git add --all$",
      "^git reset$",
      "^pwd$"
    ],
    "repositoryCache": "enabled",
    "cacheTtlOverride": {
      "datasource-docker-hub-tags": 90,
      "datasource-maven:metadata-xml": 90
    },
    "encryptedWarning": "Encrypted secrets were detected in the repository config. This feature has been deprecated. Please migrate them to this portal. See the guide at http://docs.renovatebot.com/mend-hosted/migrating-secrets/",
    "onboardingConfig": {
      "$schema": "https://docs.renovatebot.com/renovate-schema.json",
      "extends": [
        "config:recommended"
      ]
    },
    "gitIgnoredAuthors": [
      "29139614+renovate[bot]@users.noreply.github.com"
    ],
    "customizeDashboard": {
      "repoProblemsHeader": "These problems occurred while renovating this repository. [View logs](https://developer.mend.io//{{platform}}/{{repository}})."
    },
    "repositoryCacheType": "local",
    "dependencyDashboardFooter": "\n- [ ] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n",
    "dependencyDashboardHeader": "This issue lists Renovate updates and detected dependencies. Read the [Dependency Dashboard](https://docs.renovatebot.com/key-concepts/dashboard/) docs to learn more.<br>[View this repository on the Mend.io Web Portal](https://developer.mend.io/{{platform}}/{{repository}}).",
    "hostRules": [],
    "onboardingNoDeps": "enabled",
    "cacheDir": "/runner/cache",
    "onboarding": true,
    "forkProcessing": "enabled",
    "requireConfig": "required",
    "platform": "github",
    "username": "renovate[bot]",
    "repositories": [
      "karlhorky/repro-renovate-pnpm-dedupe-hanging"
    ],
    "gitAuthor": "renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>",
    "deleteConfigFile": true
  }
}

DEBUG: Resolved global extends
{
  "config": {
    "allowedEnv": [
      "GO*",
      "GRADLE_OPTS",
      "RUSTC_BOOTSTRAP"
    ],
    "description": [
      "Hopefully safe environment variables to allow users to configure."
    ]
  }
}

DEBUG: Combined config
{
  "config": {
    "allowedEnv": [
      "GO*",
      "GRADLE_OPTS",
      "RUSTC_BOOTSTRAP"
    ],
    "description": [
      "Hopefully safe environment variables to allow users to configure."
    ],
    "token": "***********",
    "hostRules": [
      {
        "hostType": "docker",
        "matchHost": "docker.io",
        "username": "mdpprod2renovate",
        "password": "***********"
      },
      {
        "hostType": "docker",
        "matchHost": "hub.docker.com",
        "username": "mdpprodrenovatedc",
        "password": "***********"
      },
      {
        "hostType": "github",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "github-tags",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "github-releases",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "pod",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "terraform-provider",
        "matchHost": "api.github.com",
        "token": "***********"
      },
      {
        "hostType": "rubygems",
        "matchHost": "github.com",
        "token": "***********"
      }
    ],
    "secrets": {},
    "extends": [
      "mergeConfidence:all-badges"
    ],
    "prFooter": "This PR was generated by [Mend Renovate](https://mend.io/renovate/). View the [repository job log](https://developer.mend.io/{{platform}}/{{repository}}).",
    "redisUrl": "redis://developer-platform-bare-metal-prd.aqffol.ng.0001.use1.cache.amazonaws.com:6379",
    "binarySource": "install",
    "logLevelRemap": [
      {
        "newLogLevel": "info",
        "matchMessage": "Cache fallback URLs"
      }
    ],
    "platformCommit": "enabled",
    "allowedCommands": [
      "^git add --all$",
      "^git reset$",
      "^pwd$"
    ],
    "repositoryCache": "enabled",
    "cacheTtlOverride": {
      "datasource-docker-hub-tags": 90,
      "datasource-maven:metadata-xml": 90
    },
    "encryptedWarning": "Encrypted secrets were detected in the repository config. This feature has been deprecated. Please migrate them to this portal. See the guide at http://docs.renovatebot.com/mend-hosted/migrating-secrets/",
    "onboardingConfig": {
      "$schema": "https://docs.renovatebot.com/renovate-schema.json",
      "extends": [
        "config:recommended"
      ]
    },
    "gitIgnoredAuthors": [
      "29139614+renovate[bot]@users.noreply.github.com"
    ],
    "customizeDashboard": {
      "repoProblemsHeader": "These problems occurred while renovating this repository. [View logs](https://developer.mend.io//{{platform}}/{{repository}})."
    },
    "repositoryCacheType": "local",
    "dependencyDashboardFooter": "\n- [ ] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n",
    "dependencyDashboardHeader": "This issue lists Renovate updates and detected dependencies. Read the [Dependency Dashboard](https://docs.renovatebot.com/key-concepts/dashboard/) docs to learn more.<br>[View this repository on the Mend.io Web Portal](https://developer.mend.io/{{platform}}/{{repository}}).",
    "onboardingNoDeps": "enabled",
    "cacheDir": "/runner/cache",
    "onboarding": true,
    "forkProcessing": "enabled",
    "requireConfig": "required",
    "platform": "github",
    "username": "renovate[bot]",
    "repositories": [
      "karlhorky/repro-renovate-pnpm-dedupe-hanging"
    ],
    "gitAuthor": "renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>",
    "deleteConfigFile": true
  }
}

DEBUG: Enabling forkProcessing while in non-autodiscover mode
DEBUG: Enabling onboardingNoDeps while in non-autodiscover mode
DEBUG: Found valid git version: 2.49.0
DEBUG: Setting global hostRules
DEBUG: Adding password authentication for docker.io (hostType=docker) to hostRules
DEBUG: Adding password authentication for hub.docker.com (hostType=docker) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-tags) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-releases) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=pod) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=terraform-provider) to hostRules
DEBUG: Adding token authentication for github.com (hostType=rubygems) to hostRules
DEBUG: Using default github endpoint: https://api.github.com/
DEBUG: Platform config
{
  "platformConfig": {
    "hostType": "github",
    "endpoint": "https://api.github.com/",
    "isGHApp": true,
    "isGhe": false
  }
  "renovateUsername": "renovate[bot]"
}

DEBUG: Using configured gitAuthor (renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>)
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: Using baseDir: /tmp/renovate
DEBUG: Using configured cacheDir: /runner/cache
DEBUG: Using containerbaseDir: /runner/cache/containerbase
DEBUG: Redis cache init
DEBUG: Redis cache connected
DEBUG: Commits limit = null
DEBUG: Setting global hostRules
DEBUG: Adding password authentication for docker.io (hostType=docker) to hostRules
DEBUG: Adding password authentication for hub.docker.com (hostType=docker) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-tags) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-releases) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=pod) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=terraform-provider) to hostRules
DEBUG: Adding token authentication for github.com (hostType=rubygems) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: validatePresets()
DEBUG: Reinitializing hostRules for repo
DEBUG: Clearing hostRules
DEBUG: Adding password authentication for docker.io (hostType=docker) to hostRules
DEBUG: Adding password authentication for hub.docker.com (hostType=docker) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-tags) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-releases) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=pod) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=terraform-provider) to hostRules
DEBUG: Adding token authentication for github.com (hostType=rubygems) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
INFO: Repository started
{
  "renovateVersion": "40.50.0"
}

DEBUG: Using localDir: /tmp/renovate/repos/github/karlhorky/repro-renovate-pnpm-dedupe-hanging
DEBUG: PackageFiles.clear() - Package files deleted
DEBUG: initRepo("karlhorky/repro-renovate-pnpm-dedupe-hanging")
DEBUG: hostRules: authentication already set for api.github.com
DEBUG: Using queue: host=api.github.com, concurrency=16
DEBUG: karlhorky/repro-renovate-pnpm-dedupe-hanging default branch = main
DEBUG: Using app token for git init
DEBUG: fetchBranchCommits(): url=https://**redacted**@github.com/karlhorky/repro-renovate-pnpm-dedupe-hanging.git
DEBUG: Repository cache is restored from revision 13
DEBUG: Resetting npmrc
DEBUG: Resetting npmrc
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: findPr(renovate/configure, Configure Renovate, !open)
DEBUG: http cache: saving https://api.github.com/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/pulls?per_page=20&state=all&sort=updated&direction=desc&page=1 (etag=W/"8be292c17b3b0067fea20392710124e32eb8b32bc9dd07cced3e1f9cffa42329", lastModified=undefined)
DEBUG: getPrList success
{
  "pullsTotal": 2
  "requestsTotal": 1
  "apiQuotaAffected": true
}

DEBUG: findPr(renovate/configure, chore: Configure Renovate, !open)
DEBUG: Checking cached config file name
DEBUG: http cache: Using cached response: https://api.github.com/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/contents/renovate.json from 2025-06-17T09:34:49.623Z
DEBUG: Existing config file confirmed
DEBUG: Repository config
{
  "fileName": "renovate.json"
  "config": {
    "extends": [
      "config:base"
    ],
    "postUpdateOptions": [
      "pnpmDedupe"
    ],
    "packageRules": [
      {
        "groupName": "dependency upgrades - non-major",
        "matchDepTypes": [
          "devDependencies",
          "packageManager"
        ],
        "updateTypes": [
          "patch",
          "minor"
        ]
      }
    ]
  }
}

DEBUG: Repo is onboarded
DEBUG: http cache: Using cached response: https://api.github.com/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/contents/renovate.json from 2025-06-17T09:34:49.623Z
DEBUG: migrateAndValidate()
DEBUG: Config migration necessary
{
  "oldConfig": {
    "extends": [
      "mergeConfidence:all-badges",
      "config:base"
    ],
    "postUpdateOptions": [
      "pnpmDedupe"
    ],
    "packageRules": [
      {
        "groupName": "dependency upgrades - non-major",
        "matchDepTypes": [
          "devDependencies",
          "packageManager"
        ],
        "updateTypes": [
          "patch",
          "minor"
        ]
      }
    ]
  }
  "newConfig": {
    "extends": [
      "mergeConfidence:all-badges",
      "config:recommended"
    ],
    "postUpdateOptions": [
      "pnpmDedupe"
    ],
    "packageRules": [
      {
        "groupName": "dependency upgrades - non-major",
        "matchDepTypes": [
          "devDependencies",
          "packageManager"
        ],
        "matchUpdateTypes": [
          "patch",
          "minor"
        ]
      }
    ]
  }
}

DEBUG: Post-massage config
{
  "config": {
    "extends": [
      "mergeConfidence:all-badges",
      "config:recommended"
    ],
    "postUpdateOptions": [
      "pnpmDedupe"
    ],
    "packageRules": [
      {
        "groupName": "dependency upgrades - non-major",
        "matchDepTypes": [
          "devDependencies",
          "packageManager"
        ],
        "matchUpdateTypes": [
          "patch",
          "minor"
        ]
      }
    ]
  }
}

DEBUG: Found repo ignorePaths
{
  "ignorePaths": [
    "**/node_modules/**",
    "**/bower_components/**",
    "**/vendor/**",
    "**/examples/**",
    "**/__tests__/**",
    "**/test/**",
    "**/tests/**",
    "**/__fixtures__/**"
  ]
}

DEBUG: http cache: saving https://api.github.com/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/dependabot/alerts?state=open&direction=asc&per_page=100 (etag="0a8e0f3a5e9612357c0ea4b0b0762faf261d6737b8808f13be199def0097319c", lastModified=undefined)
DEBUG: No vulnerability alerts found
DEBUG: No vulnerability alerts found
DEBUG: findIssue(Dependency Dashboard)
DEBUG: Issues cache: synced
DEBUG: Found issue 4
DEBUG: http cache: saving https://api.github.com/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/issues/4 (etag=W/"e1b55a1b8d13e0f3f44f95d50a5f620a478c56d4ff25dbb174ed618d347be3a7", lastModified=Tue, 17 Jun 2025 09:51:34 GMT)
DEBUG: No baseBranches
DEBUG: extract()
DEBUG: Cached extract for sha=d5bb97813b63312ff56e1333e1b65d4d88277c3b is valid and can be used
DEBUG: Deleted cached dep updates
INFO: Dependency extraction complete
{
  "baseBranch": "main"
  "stats": {
    "managers": {
      "npm": {
        "fileCount": 1,
        "depCount": 2
      }
    },
    "total": {
      "fileCount": 1,
      "depCount": 2
    }
  }
}

DEBUG: hostRules: no authentication for registry.npmjs.org
DEBUG: Using queue: host=registry.npmjs.org, concurrency=999
DEBUG: http cache: Using cached response: https://registry.npmjs.org/stylelint from 2025-06-12T16:04:32.419Z
DEBUG: http cache: Using cached response: https://registry.npmjs.org/pnpm from 2025-06-15T14:40:52.199Z
DEBUG: PackageFiles.add() - Package file saved for base branch
{
  "baseBranch": "main"
}

DEBUG: Package releases lookups complete
{
  "baseBranch": "main"
}

DEBUG: Repository libYears
{
  "libYears": {
    "managers": {
      "npm": 0.20730319295408423
    },
    "total": 0.20730319295408423
  }
  "dependencyStatus": {
    "outdated": 2,
    "total": 2
  }
}

DEBUG: branchifyUpgrades
DEBUG: detectSemanticCommits()
DEBUG: semanticCommits: returning "disabled" from cache
DEBUG: 2 flattened updates found: stylelint, pnpm
DEBUG: Returning 1 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "baseBranch": "main"
  "config": {
    "npm": [
      {
        "deps": [
          {
            "currentValue": "16.19.1",
            "currentVersion": "16.19.1",
            "currentVersionAgeInDays": 52,
            "currentVersionTimestamp": "2025-04-25T15:08:43.345Z",
            "datasource": "npm",
            "depName": "stylelint",
            "depType": "devDependencies",
            "fixedVersion": "16.19.1",
            "homepage": "https://stylelint.io",
            "isSingleVersion": true,
            "lockedVersion": "16.19.1",
            "mostRecentTimestamp": "2025-05-29T16:03:49.584Z",
            "packageName": "stylelint",
            "prettyDepType": "devDependency",
            "registryUrl": "https://registry.npmjs.org",
            "sourceUrl": "https://github.com/stylelint/stylelint",
            "versioning": "npm",
            "warnings": [],
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "16.20.0",
                "newValue": "16.20.0",
                "releaseTimestamp": "2025-05-29T16:03:49.584Z",
                "newVersionAgeInDays": 18,
                "newMajor": 16,
                "newMinor": 20,
                "newPatch": 0,
                "updateType": "minor",
                "isBreaking": false,
                "libYears": 0.09325552508244546,
                "branchName": "renovate/dependency-upgrades-non-major"
              }
            ]
          },
          {
            "commitMessageTopic": "pnpm",
            "currentValue": "10.10.0+sha512.d615db246fe70f25dcfea6d8d73dee782ce23e2245e3c4f6f888249fb568149318637dca73c2c5c8ef2a4ca0d5657fb9567188bfab47f566d1ee6ce987815c39",
            "currentVersion": "10.10.0",
            "currentVersionAgeInDays": 50,
            "currentVersionTimestamp": "2025-04-27T23:18:02.369Z",
            "datasource": "npm",
            "depName": "pnpm",
            "depType": "packageManager",
            "fixedVersion": "10.10.0+sha512.d615db246fe70f25dcfea6d8d73dee782ce23e2245e3c4f6f888249fb568149318637dca73c2c5c8ef2a4ca0d5657fb9567188bfab47f566d1ee6ce987815c39",
            "homepage": "https://pnpm.io",
            "isSingleVersion": true,
            "mostRecentTimestamp": "2025-06-08T14:21:29.623Z",
            "packageName": "pnpm",
            "prettyDepType": "packageManager",
            "registryUrl": "https://registry.npmjs.org",
            "sourceDirectory": "pnpm",
            "sourceUrl": "https://github.com/pnpm/pnpm",
            "versioning": "npm",
            "warnings": [],
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "10.12.1",
                "newValue": "10.12.1",
                "releaseTimestamp": "2025-06-08T14:21:29.623Z",
                "newVersionAgeInDays": 8,
                "newMajor": 10,
                "newMinor": 12,
                "newPatch": 1,
                "updateType": "minor",
                "isBreaking": false,
                "libYears": 0.11404766787163877,
                "branchName": "renovate/dependency-upgrades-non-major"
              }
            ]
          }
        ],
        "extractedConstraints": {
          "pnpm": "10.10.0+sha512.d615db246fe70f25dcfea6d8d73dee782ce23e2245e3c4f6f888249fb568149318637dca73c2c5c8ef2a4ca0d5657fb9567188bfab47f566d1ee6ce987815c39"
        },
        "lockFiles": [
          "pnpm-lock.yaml"
        ],
        "managerData": {
          "hasPackageManager": true,
          "npmrcFileName": null,
          "pnpmShrinkwrap": "pnpm-lock.yaml",
          "yarnZeroInstall": false
        },
        "packageFile": "package.json",
        "skipInstalls": true
      }
    ]
  }
}

DEBUG: detectSemanticCommits()
DEBUG: semanticCommits: returning "disabled" from cache
DEBUG: processRepo()
DEBUG: Processing 1 branch: renovate/dependency-upgrades-non-major
DEBUG: getBranchPr(renovate/dependency-upgrades-non-major)
DEBUG: findPr(renovate/dependency-upgrades-non-major, undefined, open)
DEBUG: 0 PRs are currently open
DEBUG: ConcurrentPRs count = 0
DEBUG: 0 already existing branches found.
DEBUG: Branches count = 0
DEBUG: Calculating PRs created so far in this hour currentHourStart=2025-06-17T10:00:00.000Z
DEBUG: 0 PRs have been created so far in this hour.
DEBUG: HourlyPRs count = 0
DEBUG: syncBranchState() (branch="renovate/dependency-upgrades-non-major")
DEBUG: getBranchPr(renovate/dependency-upgrades-non-major) (branch="renovate/dependency-upgrades-non-major")
DEBUG: findPr(renovate/dependency-upgrades-non-major, undefined, open) (branch="renovate/dependency-upgrades-non-major")
DEBUG: branchExists=false (branch="renovate/dependency-upgrades-non-major")
DEBUG: dependencyDashboardCheck=unlimit (branch="renovate/dependency-upgrades-non-major")
DEBUG: recreateClosed is true. No need to check for closed PR. (branch="renovate/dependency-upgrades-non-major")
DEBUG: Open PR Count: 0, Existing Branch Count: 0, Hourly PR Count: 0 (branch="renovate/dependency-upgrades-non-major")
DEBUG: prHourlyLimit of the upgrades present in this branch (branch="renovate/dependency-upgrades-non-major")
{
  "limits": [
    {
      "depName": "pnpm",
      "prHourlyLimit": 2
    },
    {
      "depName": "stylelint",
      "prHourlyLimit": 2
    }
  ]
}

DEBUG: Calculated lowest prHourlyLimit among the upgrades present in this branch is 2. (branch="renovate/dependency-upgrades-non-major")
DEBUG: branchConcurrentLimit of the upgrades present in this branch (branch="renovate/dependency-upgrades-non-major")
{
  "limits": [
    {
      "depName": "pnpm",
      "branchConcurrentLimit": null
    },
    {
      "depName": "stylelint",
      "branchConcurrentLimit": null
    }
  ]
}

DEBUG: Calculated lowest branchConcurrentLimit among the upgrades present in this branch is 10. (branch="renovate/dependency-upgrades-non-major")
DEBUG: Checking schedule(schedule=at any time, tz=null, now=2025-06-17T10:07:32.964Z) (branch="renovate/dependency-upgrades-non-major")
DEBUG: No schedule defined (branch="renovate/dependency-upgrades-non-major")
DEBUG: GET https://api.github.com/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/branches/main/protection = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=75) (branch="renovate/dependency-upgrades-non-major")
DEBUG: No branch protection found for main (branch="renovate/dependency-upgrades-non-major")
DEBUG: Converting rebaseWhen=auto to rebaseWhen=conflicted because no rule for behind-base-branch applies (branch="renovate/dependency-upgrades-non-major")
DEBUG: Branch needs creating (branch="renovate/dependency-upgrades-non-major")
DEBUG: Using reuseExistingBranch: false (branch="renovate/dependency-upgrades-non-major")
DEBUG: Setting current branch to main (branch="renovate/dependency-upgrades-non-major")
DEBUG: syncGit(): Initializing git repository into /tmp/renovate/repos/github/karlhorky/repro-renovate-pnpm-dedupe-hanging (branch="renovate/dependency-upgrades-non-major")
DEBUG: Performing blobless clone (branch="renovate/dependency-upgrades-non-major")
DEBUG: git clone completed (branch="renovate/dependency-upgrades-non-major")
{
  "durationMs": 362
}

DEBUG: latest repository commit (branch="renovate/dependency-upgrades-non-major")
{
  "latestCommit": {
    "hash": "d5bb97813b63312ff56e1333e1b65d4d88277c3b",
    "date": "2025-06-17T11:34:57+02:00",
    "message": "Add lockfile and .gitignore",
    "refs": "HEAD -> main, origin/main, origin/HEAD",
    "body": "",
    "author_name": "Karl Horky",
    "author_email": "karl.horky@gmail.com"
  }
}

DEBUG: Current branch SHA: d5bb97813b63312ff56e1333e1b65d4d88277c3b (branch="renovate/dependency-upgrades-non-major")
DEBUG: latest commit (branch="renovate/dependency-upgrades-non-major")
{
  "branchName": "main"
  "latestCommitDate": "2025-06-17T11:34:57+02:00"
  "sha": "d5bb97813b63312ff56e1333e1b65d4d88277c3b"
}

DEBUG: manager.getUpdatedPackageFiles() reuseExistingBranch=false (branch="renovate/dependency-upgrades-non-major")
DEBUG: npm.updateDependency(): packageManager.pnpm = 10.12.1 (branch="renovate/dependency-upgrades-non-major")
DEBUG: Updating pnpm in package.json (branch="renovate/dependency-upgrades-non-major")
DEBUG: npm.updateDependency(): devDependencies.stylelint = 16.20.0 (branch="renovate/dependency-upgrades-non-major")
DEBUG: Updating stylelint in package.json (branch="renovate/dependency-upgrades-non-major")
DEBUG: updateArtifacts for updatedPackageFiles (branch="renovate/dependency-upgrades-non-major")
DEBUG: npm.updateArtifacts(package.json) (branch="renovate/dependency-upgrades-non-major")
DEBUG: Found 0 npm host rule(s) (branch="renovate/dependency-upgrades-non-major")
DEBUG: Found 1 host rule(s) without host type (branch="renovate/dependency-upgrades-non-major")
DEBUG: Found 1 host rule(s) without host type after dropping duplicates (branch="renovate/dependency-upgrades-non-major")
DEBUG: No node constraint found - using latest (branch="renovate/dependency-upgrades-non-major")
DEBUG: Setting CONTAINERBASE_CACHE_DIR to /runner/cache/containerbase (branch="renovate/dependency-upgrades-non-major")
DEBUG: Using containerbase dynamic installs (branch="renovate/dependency-upgrades-non-major")
DEBUG: Resolved stable matching version (branch="renovate/dependency-upgrades-non-major")
{
  "toolName": "node"
  "constraint": null
  "resolvedVersion": "22.16.0"
}

DEBUG: http cache: Using cached response: https://registry.npmjs.org/corepack from 2025-06-16T13:07:35.159Z (branch="renovate/dependency-upgrades-non-major")
DEBUG: Resolved stable matching version (branch="renovate/dependency-upgrades-non-major")
{
  "toolName": "corepack"
  "resolvedVersion": "0.33.0"
}

DEBUG: Executing command (branch="renovate/dependency-upgrades-non-major")
{
  "command": "install-tool node 22.16.0"
}

DEBUG: exec completed (branch="renovate/dependency-upgrades-non-major")
{
  "durationMs": 5563
  "stdout": "[10:07:34.278] INFO (911): Installing tool node@22.16.0...\nv22.16.0\n10.9.2\n0.32.0\n[10:07:39.290] INFO (911): Install tool node succeeded in 5s.\n"
  "stderr": ""
}

DEBUG: Executing command (branch="renovate/dependency-upgrades-non-major")
{
  "command": "install-tool corepack 0.33.0"
}

DEBUG: exec completed (branch="renovate/dependency-upgrades-non-major")
{
  "durationMs": 2545
  "stdout": "[10:07:39.787] INFO (948): Installing npm corepack@0.33.0...\n0.33.0\n[10:07:41.317] INFO (948): Install tool corepack succeeded in 1.5s.\n"
  "stderr": ""
}

DEBUG: Executing command (branch="renovate/dependency-upgrades-non-major")
{
  "command": "corepack use pnpm@10.12.1"
}

DEBUG: exec completed (branch="renovate/dependency-upgrades-non-major")
{
  "durationMs": 5517
  "stdout": "Installing pnpm@10.12.1 in the project...\n\nProgress: resolved 1, reused 0, downloaded 0, added 0\nProgress: resolved 39, reused 0, downloaded 16, added 0\nProgress: resolved 89, reused 0, downloaded 85, added 0\nPackages: +113\n++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\nProgress: resolved 113, reused 0, downloaded 113, added 113, done\n\ndevDependencies:\n+ stylelint 16.20.0\n\nDone in 3.8s using pnpm v10.12.1\n"
  "stderr": ""
}

DEBUG: Returning updated package.json (branch="renovate/dependency-upgrades-non-major")
DEBUG: Updated 1 package files (branch="renovate/dependency-upgrades-non-major")
DEBUG: Getting updated lock files (branch="renovate/dependency-upgrades-non-major")
DEBUG: Writing package.json files (branch="renovate/dependency-upgrades-non-major")
{
  "packageFiles": [
    "package.json"
  ]
}

DEBUG: Writing any updated package files (branch="renovate/dependency-upgrades-non-major")
DEBUG: Writing package.json (branch="renovate/dependency-upgrades-non-major")
DEBUG: Found 0 npm host rule(s) (branch="renovate/dependency-upgrades-non-major")
DEBUG: Found 1 host rule(s) without host type (branch="renovate/dependency-upgrades-non-major")
DEBUG: Found 1 host rule(s) without host type after dropping duplicates (branch="renovate/dependency-upgrades-non-major")
DEBUG: Generating pnpm-lock.yaml for . (branch="renovate/dependency-upgrades-non-major")
DEBUG: Spawning pnpm install to create pnpm-lock.yaml (branch="renovate/dependency-upgrades-non-major")
DEBUG: No node constraint found - using latest (branch="renovate/dependency-upgrades-non-major")
DEBUG: Setting CONTAINERBASE_CACHE_DIR to /runner/cache/containerbase (branch="renovate/dependency-upgrades-non-major")
DEBUG: Using containerbase dynamic installs (branch="renovate/dependency-upgrades-non-major")
DEBUG: Resolved stable matching version (branch="renovate/dependency-upgrades-non-major")
{
  "toolName": "node"
  "constraint": null
  "resolvedVersion": "22.16.0"
}

DEBUG: Executing command (branch="renovate/dependency-upgrades-non-major")
{
  "command": "install-tool node 22.16.0"
}

DEBUG: exec completed (branch="renovate/dependency-upgrades-non-major")
{
  "durationMs": 573
  "stdout": "[10:07:47.752] INFO (1008): Installing tool node@22.16.0...\n[10:07:47.783] INFO (1008): tool already installed\n    tool: \"node\"\nv22.16.0\n10.9.2\n0.32.0\n[10:07:48.008] INFO (1008): Install tool node succeeded in 270ms.\n"
  "stderr": ""
}

DEBUG: Executing command (branch="renovate/dependency-upgrades-non-major")
{
  "command": "install-tool pnpm 10.12.1"
}

DEBUG: exec completed (branch="renovate/dependency-upgrades-non-major")
{
  "durationMs": 3438
  "stdout": "[10:07:48.396] INFO (1044): Installing npm pnpm@10.12.1...\n10.12.1\n[10:07:51.485] INFO (1044): Install tool pnpm succeeded in 3.1s.\n"
  "stderr": ""
}

DEBUG: Executing command (branch="renovate/dependency-upgrades-non-major")
{
  "command": "pnpm install --lockfile-only --ignore-scripts --ignore-pnpmfile"
}

DEBUG: exec completed (branch="renovate/dependency-upgrades-non-major")
{
  "durationMs": 838
  "stdout": "Done in 782ms using pnpm v10.12.1\n"
  "stderr": ""
}

DEBUG: Executing command (branch="renovate/dependency-upgrades-non-major")
{
  "command": "pnpm dedupe --config.ignore-scripts=true"
}

DEBUG: rawExec err (branch="renovate/dependency-upgrades-non-major")
{
  "err": {
    "cmd": "/bin/sh -c pnpm dedupe --config.ignore-scripts=true",
    "stderr": "",
    "stdout": "\u001b[?25l\u001b[36m?\u001b[39m \u001b[1mThe modules directory at \"/tmp/renovate/repos/github/karlhorky/repro-renovate-pnpm-dedupe-hanging/node_modules\" will be removed and reinstalled from scratch. Proceed?\u001b[22m \u001b[2m(Y/n)\u001b[22m \u001b[2m‣\u001b[22m \u001b[36mtrue\u001b[39m\u001b[4D",
    "options": {
      "cwd": "/tmp/renovate/repos/github/karlhorky/repro-renovate-pnpm-dedupe-hanging",
      "encoding": "utf-8",
      "env": {
        "NPM_CONFIG_CACHE": "/runner/cache/others/npm",
        "npm_config_store": "/runner/cache/others/pnpm",
        "npm_config_cache_dir": "/runner/cache/others/pnpm-cache",
        "npm_config_store_dir": "/runner/cache/others/pnpm-store",
        "HOME": "/home/ubuntu",
        "PATH": "/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
        "LC_ALL": "C.UTF-8",
        "LANG": "C.UTF-8",
        "CONTAINERBASE_CACHE_DIR": "/runner/cache/containerbase"
      },
      "maxBuffer": 10485760,
      "timeout": 900000
    },
    "signal": "SIGTERM",
    "name": "ExecError",
    "message": "Command failed: pnpm dedupe --config.ignore-scripts=true\nInterrupted by SIGTERM",
    "stack": "ExecError: Command failed: pnpm dedupe --config.ignore-scripts=true\nInterrupted by SIGTERM\n    at ChildProcess.<anonymous> (/usr/local/renovate/lib/util/exec/common.ts:111:11)\n    at ChildProcess.emit (node:events:530:35)\n    at ChildProcess.emit (node:domain:489:12)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)"
  }
  "durationMs": 900034
}

DEBUG: exec interrupted by SIGTERM - run needs to be aborted (branch="renovate/dependency-upgrades-non-major")
{
  "err": {
    "cmd": "/bin/sh -c pnpm dedupe --config.ignore-scripts=true",
    "stderr": "",
    "stdout": "\u001b[?25l\u001b[36m?\u001b[39m \u001b[1mThe modules directory at \"/tmp/renovate/repos/github/karlhorky/repro-renovate-pnpm-dedupe-hanging/node_modules\" will be removed and reinstalled from scratch. Proceed?\u001b[22m \u001b[2m(Y/n)\u001b[22m \u001b[2m‣\u001b[22m \u001b[36mtrue\u001b[39m\u001b[4D",
    "options": {
      "cwd": "/tmp/renovate/repos/github/karlhorky/repro-renovate-pnpm-dedupe-hanging",
      "encoding": "utf-8",
      "env": {
        "NPM_CONFIG_CACHE": "/runner/cache/others/npm",
        "npm_config_store": "/runner/cache/others/pnpm",
        "npm_config_cache_dir": "/runner/cache/others/pnpm-cache",
        "npm_config_store_dir": "/runner/cache/others/pnpm-store",
        "HOME": "/home/ubuntu",
        "PATH": "/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
        "LC_ALL": "C.UTF-8",
        "LANG": "C.UTF-8",
        "CONTAINERBASE_CACHE_DIR": "/runner/cache/containerbase"
      },
      "maxBuffer": 10485760,
      "timeout": 900000
    },
    "signal": "SIGTERM",
    "name": "ExecError",
    "message": "Command failed: pnpm dedupe --config.ignore-scripts=true\nInterrupted by SIGTERM",
    "stack": "ExecError: Command failed: pnpm dedupe --config.ignore-scripts=true\nInterrupted by SIGTERM\n    at ChildProcess.<anonymous> (/usr/local/renovate/lib/util/exec/common.ts:111:11)\n    at ChildProcess.emit (node:events:530:35)\n    at ChildProcess.emit (node:domain:489:12)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)"
  }
}

DEBUG: Passing TEMPORARY_ERROR error up (branch="renovate/dependency-upgrades-non-major")
INFO: Temporary error - aborting
DEBUG: Repository result: temporary-error, status: onboarded, enabled: true, onboarded: true
DEBUG: Repository timing splits (milliseconds)
{
  "splits": {
    "init": 2961,
    "extract": 357,
    "lookup": 1595,
    "onboarding": 1
  }
  "total": 924660
}

DEBUG: Package cache statistics
{
  "get": {
    "count": 5,
    "avgMs": 56,
    "medianMs": 23,
    "maxMs": 175,
    "totalMs": 280
  }
  "set": {
    "count": 0,
    "avgMs": 0,
    "medianMs": 0,
    "maxMs": 0,
    "totalMs": 0
  }
}

DEBUG: HTTP statistics
{
  "hosts": {
    "api.github.com": {
      "count": 7,
      "reqAvgMs": 128,
      "reqMedianMs": 114,
      "reqMaxMs": 317,
      "queueAvgMs": 2,
      "queueMedianMs": 0,
      "queueMaxMs": 11
    },
    "registry.npmjs.org": {
      "count": 3,
      "reqAvgMs": 239,
      "reqMedianMs": 136,
      "reqMaxMs": 560,
      "queueAvgMs": 0,
      "queueMedianMs": 0,
      "queueMaxMs": 1
    }
  }
  "requests": 10
}

DEBUG: HTTP cache statistics
{
  "https://api.github.com": {
    "/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/contents/renovate.json": {
      "hit": 2,
      "miss": 0
    },
    "/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/dependabot/alerts": {
      "hit": 0,
      "miss": 1
    },
    "/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/issues/4": {
      "hit": 0,
      "miss": 1
    },
    "/repos/karlhorky/repro-renovate-pnpm-dedupe-hanging/pulls": {
      "hit": 0,
      "miss": 1
    }
  }
  "https://registry.npmjs.org": {
    "/corepack": {
      "hit": 1,
      "miss": 0,
      "localMiss": 1
    },
    "/pnpm": {
      "hit": 1,
      "miss": 0,
      "localMiss": 1
    },
    "/stylelint": {
      "hit": 1,
      "miss": 0,
      "localMiss": 1
    }
  }
}

DEBUG: Lookup statistics
{
  "npm": {
    "count": 2,
    "avgMs": 1011,
    "medianMs": 1302,
    "maxMs": 1302,
    "totalMs": 2021
  }
}

INFO: Cache fallback URLs
{
  "count": 0
  "hits": {}
  "oldLevel": "debug"
}

INFO: Repository finished
{
  "cloned": true
  "durationMs": 924660
}
```

</details>

Any ideas?

rarkins commented:
@karlhorky does it only hang if you have that packageRules grouping?

karlhorky commented:
I believe the hanging (`2. "silently failing to create new PRs"`) only occurs with the grouping 

But `1. "errors in PRs"` still occur without the grouping - @peterhirn's examples do not use grouping and show that PRs are created with the `temporary error` message

karlhorky committed:
Merge branch 'main' into revert-35807-fix/pnpm/cache-dir

karlhorky committed:
Merge branch 'main' into revert-35807-fix/pnpm/cache-dir

karlhorky committed:
Merge branch 'main' into revert-35807-fix/pnpm/cache-dir

rarkins commented:
So are these problems only when pnpm itself is being upgraded?

rarkins commented:
Removing pnpmDedupe from the reproduction let it complete successfully. Both reproduction repos work if the `pnpmDedupe` option excludes pnpm itself, e.g. `{ "matchPackageNames": ["!pnpm"], "postUpdateOptions": ["pnpmDedupe"] },`

karlhorky commented:
> Removing pnpmDedupe from the reproduction let it complete successfully. Both reproduction repos work if the `pnpmDedupe` option excludes pnpm itself, e.g. `{ "matchPackageNames": ["!pnpm"], "postUpdateOptions": ["pnpmDedupe"] },`

Oh nice, so should that be the default configuration of `pnpmDedupe` in the Renovate docs?

Unusual that the `pnpmDedupe` stopped working recently - I added it in May 2024 and I haven't seen issues with it since then.

karlhorky commented:
Update: I tried [committing invalid JSON in `renovate.json`](https://github.com/upleveled/portfolio-cms/commit/8b1e72e0d6346fcc8969c02dca5ad37231998302) in the hope this would invalidate the bad Renovated state causing the hanging with Shared Config Presets, but after [fixing the JSON error again](https://github.com/upleveled/portfolio-cms/commit/373c13f8648ff22f4d143c16f64728d7b4a67e56) this did not resolve the problem, and the hanging happened again afterwards: 

1. Renovate opened an issue reporting the JSON error https://github.com/upleveled/portfolio-cms/issues/177
2. Back to `pnpm dedupe` hanging behavior afterwards
3. Now the JSON issue is also not being closed in this repo anymore 🤪

karlhorky commented:
Update 2: I tried to [change the `"extends"` value to `"config:base"`](https://github.com/upleveled/portfolio-cms/commit/67b0437d0187c2083646df53c62bed123c5bee49), to ignore my shared config, with the hope that a successful run + PR merge would cause Renovate to forget its retained bad state for the repo / Shared Config Presets, but after [merging one of the created PRs](https://github.com/upleveled/portfolio-cms/pull/179) and [changing it back to `"extends": ["github>karlhorky/renovate-config:default.json5"]`](https://github.com/upleveled/portfolio-cms/commit/aacd30b166b6fd105f4e2ae8fd2ebe243628a5c8), the `pnpm dedupe` error and the hanging behavior returned 😔

![Screenshot_20250620-080010](https://github.com/user-attachments/assets/764385f7-bb8f-42fe-a614-a715451d1504)

![Screenshot_20250620-080450](https://github.com/user-attachments/assets/00da7c6d-94d3-4d13-85d6-aa28ead6f5a6)

@rarkins @viceice @HonkingGoose is there a way for me to manually reset the bad saved state that Renovate appears to have about the repo / Shared Config Preset? Or maybe the suggested `pnpmDedupe` fix from @rarkins won't work for this Shared Config Preset?

My Shared Config Preset here: 

- https://github.com/karlhorky/renovate-config/blob/main/default.json5

karlhorky commented:
Another thing I could imagine trying is to remove the content of the pnpm lockfile, while leaving the file - maybe that would be a workaround? 🤔

karlhorky commented:
## Current state: `pnpmDedupe` can lead to Renovate hanging or `temporary-error`

For anyone running into this `pnpm dedupe` error in the meantime

It appears [the `"postUpdateOptions": ["pnpmDedupe"]` feature of Renovate](https://docs.renovatebot.com/configuration-options/#postupdateoptions) doesn't currently work in some situations with Shared Config Presets and can lead to hanging behavior (see my posts above) or a `temporary-error` artifact update problem, possibly related to some bad saved state or pnpm `store-dir` misconfiguration in Renovate

I disabled the `"postUpdateOptions": ["pnpmDedupe"]` feature for now: 

- https://github.com/karlhorky/renovate-config/commit/e514364038ee4543af3854dcbb3a9b2183f7f078

Maybe running the `pnpm dedupe` (or `pnpm dedupe --config.ignore-scripts=true` command) manually in a GitHub Actions CI step on all Renovate PRs is the way forward for now... 🤔
